### PR TITLE
guid-unmigration: add unit tests for utility functions and duplicate resource identification

### PR DESCRIPTION
**Pull request by nflynt**
## Testing
Adds unit tests for several utility functions, as well as identifyCRTBs, identifyPRTBs and identifyGRBs, improving overall coverage. These test only business logic, and require no network calls or external state, so they should be quite lightweight.

### Automated Testing
* Test types added/modified:
    * Unit

### CODE REVIEW

File: pkg/agent/clean/adunmigration/ldap_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			wantLoggedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			result := isGUID(test.principalId)
+			if result != test.wantResult {
+				t.Errorf("expected isGUID to be %v, but got %v", test.wantResult, result)
```
Comment (pmatseykanets) [inline pkg/agent/clean/adunmigration/ldap_test.go:62]: Q: Here and in other places: should this be an assert if the `assert` package is being used? Alternatively asserts can be become idiomatic `t.{Errorf|t.Fatalf}`s.

File: pkg/agent/clean/adunmigration/ldap_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	logrusTest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEscapeUUID(t *testing.T) {
```
Comment (pmatseykanets) [inline pkg/agent/clean/adunmigration/ldap_test.go:11]: Q: Do we want this and the next test to run in parallel by any chance? As it stands right now all other tests with `t.Parallel()` in this package will be paused until these two are finished.

Comment (nflynt) [reply pkg/agent/clean/adunmigration/ldap_test.go:11]: For `TestIsGUID` we can't use parallel because logrus state is global, and the hook assertion becomes a race condition. I'll add a note to clarify. For `TestEscapeUUID` I just missed it :)

**Review by crobby — COMMENTED** [review]
I really like what is here.  Any thoughts about maybe adding tests for findLdapUser and updateADConfigMigrationStatus?  I think updateADConfigMigrationStatus might require a refactor if we do want to test it reasonably.
Also, if we had tests for migrate*RTB or maybe even just update*RTB|Token, that would greatly improve the coverage.  They might not be so hard to test since they are now passed just the interface that we care about rather than the whole scaled context.
